src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderConfig.tsx

```diff
@@ -276,41 +236,177 @@ export const LimitOrderConfig = ({
+          {' '}
+          {priceAsset.symbol}
+        </CText>
+        <Text as='span' color='text.subtle' translation='limitOrder.explanation.orderWill.prefix' />
+        <CText as='span' color='white' fontWeight='medium'>
+          {' '}
+          {formattedSellAmount}
+        </CText>
+        <CText as='span' color='white' fontWeight='medium'>
+          {' '}
+          {sellAsset.symbol}
+        </CText>
+        <Text as='span' color='text.subtle' translation='limitOrder.explanation.ensuring.prefix' />
+        <CText as='span' color='white' fontWeight='medium'>
+          {' '}
+          {formattedReceiveAmount}
+        </CText>
+        <CText as='span' color='white' fontWeight='medium'>
+          {' '}
+          {buyAsset.symbol}
+        </CText>
+        <Text as='span' color='text.subtle' translation='limitOrder.explanation.ensuring.suffix' />
+      </CText>
+    )
+  }, [
+    isLoading,
+    sellAmountCryptoPrecision,
+    limitPriceForSelectedPriceDirection,
+    priceDirection,
+    sellAsset.symbol,
+    buyAsset.symbol,
+    priceAsset.symbol,
+    receiveAmount,
+  ])
+
   return (
     <Stack spacing={4} px={6} py={4}>
       <Flex justifyContent='space-between' alignItems='center'>
         <HStack>
-          <Text translation='limitOrder.whenPriceReaches' />
-          {renderDelta}
+          <Text translation='limitOrder.when' />
+          <Button
+            variant='unstyled'
+            onClick={handleTokenTextClick}
+            fontWeight='bold'
+            fontSize='sm'
+            position='relative'
+            _after={linkAfter}
+            _hover={linkHover}
+          >
+            1 {displayAsset.symbol}
+          </Button>
+          <Text translation='limitOrder.isWorth' />
         </HStack>
         <Flex justifyContent='space-between' alignItems='center'>
-          <Text translation='limitOrder.expiry' mr={4} />
-          <Menu isLazy>
-            <MenuButton as={Button} rightIcon={timePeriodRightIcon} isDisabled={isLoading}>
-              <Text translation={expiryOptionTranslation} />
-            </MenuButton>
-            <MenuList zIndex='modal'>
-              <MenuOptionGroup type='radio' value={expiry} onChange={handleChangeExpiryOption}>
-                {expiryOptions}
-              </MenuOptionGroup>
-            </MenuList>
-          </Menu>
+          <Text translation='limitOrder.market' mr={2} />
+          <Button
+            variant='unstyled'
+            onClick={handleSetMarketLimit}
+            isDisabled={isMarketButtonDisabled}
+            fontWeight='medium'
+            fontSize='sm'
+            position='relative'
+            _after={linkAfter}
+            _hover={linkHover}
+            opacity={isMarketButtonDisabled ? 0.5 : 1}
+            cursor={isMarketButtonDisabled ? 'not-allowed' : 'pointer'}
+          >
+            {bnOrZero(marketPriceBuyAsset).toFixed(6)} {priceAsset.symbol}
```

Comment:
blocking: this will only work in one direction. This should not be the market price of buy asset, but the rate for the given direction.
e.g currently when selling 1 USDC
- if I'm on the "When 1 FOX" side, this should be worth ~0.025877 USDC (amount of USDC per 1 FOX)
- if I'm on the "When 1 USDC" side, this should be worth ~38.64 FOX (amount of FOX per 1 USDC)

Reply:
Makes sense, fixed in [e905124](https://github.com/shapeshift/web/pull/8970/commits/e9051249766cda9ac47762f1ce453ba3f1e35c9f)

Reply:
Confirmed fixed :fridaydog: https://jam.dev/c/7ae1a276-0190-4d6c-8736-ee27df6b80e6